### PR TITLE
Richard_Reverse

### PR DESCRIPTION
**Pull request by RichardSeverich**

### CODE REVIEW

File: src/main/java/org/fundacionjala/coding/richard/Reverse.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+ * String and returns the same string in reverse order
+ */
+public class Reverse {
+
+    private boolean flag = true;
+
+    /**
+     * This method is constructor.
+     */
+    public Reverse() {
```
Comment (carledriss) [inline src/main/java/org/fundacionjala/coding/richard/Reverse.java]: Remove this unnecessary constructor

File: src/main/java/org/fundacionjala/coding/richard/Reverse.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+     *
+     * @param cad this variable contains a string.
+     * @return reverse string.
+     */
+    public String reverse(String cad) {
+        if (flag) {
+            cad = new StringBuilder(cad).reverse().toString();
+            flag = false;
+        }
+        return cad.length() > 0 ? cad.substring(0, cad.length()) : String.valueOf(cad.toCharArray()[cad.length() - 1]);
```
Comment (carledriss) [inline src/main/java/org/fundacionjala/coding/richard/Reverse.java]: according to the description this should be a recursive method.

File: src/main/java/org/fundacionjala/coding/richard/Reverse.java (hunk 1 above)
Comment (RichardSeverich) [reply src/main/java/org/fundacionjala/coding/richard/Reverse.java]: ok.

File: src/main/java/org/fundacionjala/coding/richard/Reverse.java (hunk 2 above)
Comment (RichardSeverich) [reply src/main/java/org/fundacionjala/coding/richard/Reverse.java]: ok I solved that problem.

**Comment by carledriss** [comment]
LGTM!

**Pull request merged**